File: CoreEmulation/android/input/InputManager.cpp
#include "../../input/InputManager.h"

#include <iostream>

namespace Pure3X
{

bool InputManager::m_initialized = false;

bool InputManager::Initialize()
{
    std::cout << "[Input] Inicializando...\n";

    m_initialized = true;

    std::cout << "[OK] Input Manager inicializado.\n";
    return true;
}

void InputManager::Update()
{
    if (!m_initialized)
        return;

    // Entrada Android será implementada futuramente.
}

void InputManager::Shutdown()
{
    if (!m_initialized)
        return;

    std::cout << "[Input] Encerrando...\n";

    m_initialized = false;

    std::cout << "[OK] Input Manager finalizado.\n";
}

} // namespace Pure3X

File: AndroidRuntime/app/src/main/java/InputManager.java
package com.pure3x.engine;

public class InputManager {

    private boolean initialized;
    private String controller;
    private boolean vibration;
    private int connectedControllers;

    public InputManager() {
        initialized = false;
        controller = "Touch Screen";
        vibration = true;
        connectedControllers = 1;
    }

    public void initialize() {

        initialized = true;

        System.out.println("==================================");
        System.out.println("Inicializando Entrada...");
        System.out.println("Controle: " + controller);
        System.out.println("Controles conectados: " + connectedControllers);
        System.out.println("Vibração: " + (vibration ? "Ligada" : "Desligada"));
        System.out.println("Input Manager pronto.");
        System.out.println("==================================");

    }

    public boolean isInitialized() {
        return initialized;
    }

    public void shutdown() {
        initialized = false;
        System.out.println("Input Manager desligado.");
    }

    public void setController(String value) {
        controller = value;
    }

    public String getController() {
        return controller;
    }

    public void setConnectedControllers(int value) {
        connectedControllers = value;
    }

    public int getConnectedControllers() {
        return connectedControllers;
    }

    public void setVibration(boolean value) {
        vibration = value;
    }

    public boolean isVibrationEnabled() {
        return vibration;
    }

    public String getStatus() {

        if (initialized) {
            return "Entrada ativa";
        }

        return "Entrada parada";
    }

}
